src/main/java/org/openstreetmap/atlas/geography/atlas/BareAtlas.java

```diff
@@ -514,18 +516,30 @@ public SnappedEdge snapped(final Location point, final Distance threshold)
     {
         final Time begin = Time.now();
 
+        final Supplier<Iterable<Node>> nodesWithin = getCachingSupplier(nodesWithin(boundary),
+                ItemType.NODE);
+        final Supplier<Iterable<Edge>> edgesIntersecting = getCachingSupplier(
+                edgesIntersecting(boundary), ItemType.EDGE);
+        final Supplier<Iterable<Area>> areasIntersecting = getCachingSupplier(
+                areasIntersecting(boundary), ItemType.AREA);
+        final Supplier<Iterable<Line>> linesIntersecting = getCachingSupplier(
```

Comment:
Wondering if we can add the same caching strategy to the Predicate subAtlas call.

Reply:
If the filtering is expensive and we re-filter many times I'd say yes. But we'd want to run the profiler on a big atlas first to make sure it is a worthwhile change.